components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1805,1229 +1180,654 @@ subroutine tphysac (ztodt,   cam_in,  &
 
-    call cloud_diagnostics_calc(state, pbuf)
+  !-----------------------------------------------------------------------------
+  ! Moist physical parameteriztions complete: 
+  ! send dynamical variables, and derived variables to history file
+  !-----------------------------------------------------------------------------
+  call t_startf('bc_history_write')
+  call diag_phys_writeout(state, cam_out%psl)
+  call diag_conv(state, ztodt, pbuf)
+  call t_stopf('bc_history_write')
 
-    call t_stopf('bc_cld_diag_history_write')
+  !-----------------------------------------------------------------------------
+  ! Write cloud diagnostics on history file
+  !-----------------------------------------------------------------------------
+  call t_startf('bc_cld_diag_history_write')
+  call cloud_diagnostics_calc(state, pbuf)
+  call t_stopf('bc_cld_diag_history_write')
 
-if (l_rad) then
-    !===================================================
-    ! Radiation computations
-    !===================================================
+  !-----------------------------------------------------------------------------
+  ! Radiation computations
+  !-----------------------------------------------------------------------------
+  if (l_rad) then
     call t_startf('radiation')
-
-
-    call radiation_tend(state,ptend, pbuf, &
-         cam_out, cam_in, &
-         cam_in%landfrac,landm,cam_in%icefrac, cam_in%snowhland, &
-         fsns,    fsnt, flns,    flnt,  &
-         fsds, net_flx,is_cmip6_volc, ztodt, clear_rh=mmf_clear_rh)
+    call radiation_tend(state,ptend, pbuf, cam_out, cam_in, &
+                        cam_in%landfrac, landm, cam_in%icefrac, cam_in%snowhland, &
+                        fsns, fsnt, flns, flnt, fsds, &
+                        net_flx, is_cmip6_volc, ztodt, clear_rh=mmf_clear_rh)
 
     ! Set net flux used by spectral dycores
     do i=1,ncol
-       tend%flx_net(i) = net_flx(i)
+      tend%flx_net(i) = net_flx(i)
     end do
     
-    ! don't add radiative tendency to GCM temperature in case of superparameterization
-    ! as it was added above as part of crm tendency.
-    if (use_MMF) ptend%s = 0.
+    ! For MMF, don't add radiative tendency to GCM as it was added in CRM
+    ptend%s = 0.
 
     call physics_update(state, ptend, ztodt, tend)
-    
-    !call check_energy_chng(state, tend, "radheat", nstep, ztodt, zero, zero, zero, net_flx)
-    if (use_MMF) then
-      call check_energy_chng(state, tend, "spradheat", nstep, ztodt, zero, zero, zero, zero) 
-    else 
-      call check_energy_chng(state, tend, "radheat", nstep, ztodt, zero, zero, zero, net_flx)
-    endif
-
+    call check_energy_chng(state, tend, "mmfradheat", nstep, ztodt, zero, zero, zero, zero) 
     call t_stopf('radiation')
 
-end if ! l_rad
+  end if ! l_rad
 
-    if(do_aerocom_ind3) then
-       call cloud_top_aerocom(state, pbuf) 
-    end if
-
-    ! Diagnose the location of the tropopause and its location to the history file(s).
```

Comment:
I think we should put at least one timer back in to encapsulate all of this. If not the individual timers, maybe just one for "tphysbc_diagnostics" or something.

Reply:
ok, I like "tphysbc_diagnostics". I'll add it.

Reply:
done.